components/pluralrules/src/operands.rs

```diff
@@ -70,6 +71,27 @@ pub struct PluralOperands {
     pub t: u64,
 }
 
+/// PartialEq is implemented explicitly instead of derived, to account for float comparison in the
+/// internal representation.
+impl PartialEq for PluralOperands {
+    fn eq(&self, other: &Self) -> bool {
+        let exact = self.i == other.i
+            && self.v == other.v
+            && self.w == other.w
+            && self.f == other.f
+            && self.t == other.t;
+        let zero = self.n == 0_f64;
+
+        exact
+            && ((zero && self.n == other.n)
+                || (!zero && {
+                    let abs_diff = (other.n - self.n).abs();
+                    let rel_diff = abs_diff / self.n.abs(); // self.n.abs() != 0.0
+                    rel_diff < 1e-6
+                }))
+    }
```

Comment:
Optional:

Did you consider

```rust
let zero = self.n == 0_f64 && self.other == 0_f64;
exact && (zero || (self.n - other.n).abs() < f64::EPSILON)
```

Reply:
Looks nicer indeed, except I think you'd want to consider the relative error between `self.n` and `other.n`, else you may claim inequality on high-magnitude numbers, or claim equality on low magnitude numbers.  #til about `f64::EPSILON`.

And as I noted earlier, we could omit comparison of `n` altogether if we can squint and assume other fields are populated correctly.,

Reply:
...and apparently, this suggestion is 6% slower... so I'll leave as is.